Mathlib/Analysis/Calculus/PartialDerivativeTest.lean

```diff
@@ -0,0 +1,334 @@
+        coe_coe, hessianLinearCompanion, LinearMap.coe_mk, AddHom.coe_mk,
+        LinearMap.add_apply] at had₀ had₁ ⊢
+    exact (Mathlib.Tactic.Ring.add_pf_add_overlap had₀.symm had₁.symm).symm
+  map_smul' := fun m x => LinearMap.ext_iff.mpr fun x₁ => by
+    have hsm₀ := (iteratedFDeriv ℝ 2 f x₀).map_update_smul' ![x,x₁] 0 m x
+    have hsm₁ := (iteratedFDeriv ℝ 2 f x₀).map_update_smul' ![x₁,x] 1 m x
+    have h := CancelDenoms.add_subst hsm₀.symm hsm₁.symm
+    repeat rw [update₀, update₁] at h
+    exact h.symm}
+
+/-- The second iterated Frechét derivative as a quadratic map. -/
+noncomputable def iteratedFDerivQuadraticMap {V : Type*} [NormedAddCommGroup V]
+    [NormedSpace ℝ V] (f : V → ℝ) (x₀ : V) : QuadraticMap ℝ V ℝ := {
+  toFun := fun y => iteratedFDeriv ℝ 2 f x₀ ![y,y]
+  exists_companion' := by
+    use hessianBilinearCompanion f x₀
+    intro x y
+    have had := (iteratedFDeriv ℝ 2 f x₀).map_update_add'
+    have had₀ := had ![x, x + y] 0 x y
+    have had₁ := had ![x,x] 1 x y
+    have had₂ := had ![y,x] 1 x y
+    repeat rw [update₀] at had₀; rw [update₁] at had₁ had₂
+    simp only [MultilinearMap.toFun_eq_coe, coe_coe, hessianBilinearCompanion, LinearMap.coe_mk,
+      AddHom.coe_mk, hessianLinearCompanion] at had₀ had₁ had₂ ⊢
+    linarith
+  toFun_smul := fun u v => by
+    have hsm₀ := (iteratedFDeriv ℝ 2 f x₀).map_update_smul' ![v, v] 0 u v
+    have hsm₁ := (iteratedFDeriv ℝ 2 f x₀).map_update_smul' ![u • v,v] 1 u v
+    simp only [update₀, update₁, MultilinearMap.toFun_eq_coe, coe_coe, smul_eq_mul] at hsm₀ hsm₁
+    rw [smul_eq_mul, mul_assoc, ← hsm₀, hsm₁]}
+
+/-- An everywhere positive function `f:ℝⁿ → ℝ` (`n > 0`) achieves its minimum on the sphere. -/
+lemma sphere_min_of_pos_of_nonzero {V : Type*} [NormedAddCommGroup V] [NormedSpace ℝ V]
+    [FiniteDimensional ℝ V] [Nontrivial V] {f : V → ℝ} (hf : Continuous f)
+    (hf' : ∀ x ≠ 0, f x > 0) : ∃ x : Metric.sphere 0 1, ∀ y : Metric.sphere 0 1, f x.1 ≤ f y.1 := by
+  have h₀ : HasCompactSupport fun (x : (Metric.sphere (0:V) 1)) => f x := by
+    rw [hasCompactSupport_def, Function.support]
+    have : {x : Metric.sphere (0:V) (1:ℝ) | f x.1 ≠ 0} = Set.univ := by
+      apply subset_antisymm
+      simp only [ne_eq, Set.subset_univ]
+      intro a ha
+      suffices f a > 0 by aesop
+      apply hf'
+      intro hc
+      simpa [hc] using norm_eq_of_mem_sphere a
+    rw [this]
+    simp only [isClosed_univ, IsClosed.closure_eq]
+    exact CompactSpace.isCompact_univ
+  have : Nonempty (Metric.sphere (0:V) 1) :=
+    (NormedSpace.sphere_nonempty.mpr (by simp)).to_subtype
+  have ⟨m,hm⟩ := (hf.comp' continuous_subtype_val).exists_forall_le_of_hasCompactSupport h₀
+  use m
+
+/-- A continuous multilinear map is bilinear. -/
+noncomputable def continuousBilinearMap_of_continuousMultilinearMap {V : Type*}
+    [NormedAddCommGroup V] [NormedSpace ℝ V] [FiniteDimensional ℝ V]
+    (g : ContinuousMultilinearMap ℝ (fun _ : Fin 2 => V) ℝ) : V →L[ℝ] V →L[ℝ] ℝ := {
```

Comment:
If you really need this, I would just turn [ContinuousLinearMap.uncurryBilinear](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Analytic/Linear.html#ContinuousLinearMap.uncurryBilinear) into an equivalence.